### ISSUE TITLE
Adopt Nightly Build+platform and tests+platform pipelines scheduling for optimum executor

### ISSUE THREAD

**Issue by archanaNogriya** [AdoptOpenJDK/openjdk-infrastructure#1135]
**Vision:** How best optimize to run these pipelines when build and test machines are free, which makes a more efficient way to run these build pipelines. So if we can be more specific to the timeline we run the nightly builds.

**Action:**  We all need to agree to the right solution on this as a community.

**Findings from my Investigation:** 
Each Pipeline is scheduled on a daily bases,
The configuration for each pipeline is set to @daily. As per the flag,

-  OpenJDK 11 pipeline scheduled for everyday at Europ time 20:06 pm and expect to finish the next day early morning at 4.09 am, which is as per UK time 19:06 pm-3.09am GMT.

- OpenJDk8 pipelines are scheduled morning 5:26 am- next day 9:11am. which means 8 runs after 9 hours difference to openJDk 11.

However, the issue arises when these build does not start or finish at the scheduled time.

AdoptOpenJDK have build Pipelines calendar which is very handy to see the scheduled jobs,
https://ci.adoptopenjdk.net/view/Build%20and%20Test%20Pipeline%20Calendar/#date=2020-01-30|view=week-view
How we might be able to fix this?

Jenkins has predefined aliases to schedule build:
@hourly, @daily, @weekly, @monthly, @midnight
@hourly --> Build every hour at the beginning of the hour --> 0 * * * *
@daily, @midnight --> Build every day at midnight --> 0 0 * * *
@weekly --> Build every week at midnight on Sunday morning --> 0 0 * * 0
@monthly --> Build every month at midnight of the first day of the month --> 0 0 1 * *

Jenkins used a cron expression, and the different fields are:
MINUTES Minutes in one hour (0-59)
HOURS Hours in one day (0-23)
DAYMONTH Day in a month (1-31)
MONTH Month in a year (1-12)
DAYWEEK Day of the week (0-7) where 0 and 7 are Sunday
Example
If you want to schedule your build every 5 minutes, this will do the job : */5 * * * *
If you want to schedule your build every day at 8h00, this will do the job : 0 8 * * *

To allow periodically scheduled tasks to produce even load on the system, the symbol H (for “hash”) should be used wherever possible.
For example, using 0 0 * * * for a dozen daily jobs will cause a large spike at midnight. In contrast, using H H * * * would still execute each job once a day, but not all at the same time, better using limited resources.

**The solution** might be to change the configuration https://github.com/AdoptOpenJDK/openjdk-build/blob/master/pipelines/build/common/build_base_file.groovy

H(0-30) 23 * * * - OpenJDK11 (Average takes full run 8 hrs 3 min)
That means by approximately by 8am UK time It should complete.

H(0-30) 3 * * * - OpenJDK8 (Average takes full run 6 hrs 39 min)
That means by approximately 9am UK time It should complete.

I understand we need a great discussion in this and see what best we can fix with.

**Comment by sxa555** [comment]
What about the 14 and 15 pipelines?

**Comment by archanaNogriya** [comment]
Absolutely we are talking about here All tye current and future nightly build. At the moment 14 nightly builds. I started with  8 and 11 to find what is the best option we can go with. As we go further discussion I think we will have a much better understanding of how we go forward for all 14 15.. so on.

At the moment  OpenJDK14 nightly build runs at UK time 12:26 pm and Europ time 1:36 pm. again I think if we can have similar thinking how we can run these in better times when machines are free. and how we can distribute them between 8, 11 and 14 so there is no overlap.

**Comment by archanaNogriya** [comment]
Similarly, OpenJDK14 nightly build runs at UK time 00:05BST where UK time is midnight  01:05 Europ time.

**Comment by archanaNogriya** [comment]
**AdoptOpenJDK Nightly build schedules as per BST time.**
OpenJDK 8    |    OpenJDK 11      |    OpenJDK14   |    OpenJDK15
 5:26 am         |    19:06 pm           |    12:26pm         |    00:05am

**Comment by sxa555** [comment]
The formatting of that table needs an update :-) Those are the existing times - can you propose an equivalent schedule for review which has all four releases listed please?

**Comment by archanaNogriya** [comment]
> The formatting of that table needs an update :-) Those are the existing times - can you propose an equivalent schedule for review which has all four releases listed please?

@sxa555 Yes They are, The new time I have already suggested in my above comment if you see there. 
H(0-30) 23 * * * - OpenJDK11 (Average takes full run 8 hrs 3 min)
That means by approximately by 8am UK time It should complete.

H(0-30) 3 * * * - OpenJDK8 (Average takes full run 6 hrs 39 min)
That means by approximately 9am UK time It should complete.

I think that is the discussion point, when is the best time when machines are much less busy?
Here is what I suggested earlier may be more readable manner :),

**AdoptOpenJDK Suggested Nightly build schedules as per BST time.**
OpenJDK 8 ---- | ---- OpenJDK 11 ---- | ---- OpenJDK14 ---- | ---- OpenJDK15
3:30 am ----       | ---- 11:30 pm ----       | ---- 20:26pm ---- | ---- 17:05pm

**Comment by sxa555** [comment]
Thanks - I hadn't seen the 2026 and 1705 times previously

**Comment by sxa555** [comment]
Reposting the table with proposed times with correct formatting and GMT 24h timestamps:

JDK8 | JDK11 | JDK14 | JDK15
---|---|---|---
0330 | 2330 | 2026 | 1705

Maintenance / patching: 1030
